kubernetes/cache/cache.go

```diff
@@ -197,7 +197,12 @@ func (c *kialiCacheImpl) CheckNamespace(namespace string) bool {
 	if !c.isCached(namespace) {
 		return false
 	}
-	if _, exist := c.nsCache[namespace]; !exist {
+
+	c.cacheLock.RLock()
+	_, isNsCached := c.nsCache[namespace]
+	c.cacheLock.RUnlock()
```

Comment:
Between this RUnlock() and the next !isNsCached there's a possibility of schedulling to write the cache. It would probably just make sense to keep the writelock and then unlock it as soon as possible (without defer).

Reply:
The write is a one time operation with a high cost, so this fix aims to protect the read on this method.

So, in a situation as you proposed:

```
No cache existing:

T1:
  c.cacheLock.RLock()
	_, isNsCached := c.nsCache[namespace]
	c.cacheLock.RUnlock()

T1 stop
T2:
	c.cacheLock.RLock()
	_, isNsCached := c.nsCache[namespace]
	c.cacheLock.RUnlock()

	if !isNsCached {
		defer c.cacheLock.Unlock()
		c.cacheLock.Lock()
		return c.createCache(namespace)
T2 quits
T1 continues:
	if !isNsCached {
		defer c.cacheLock.Unlock()
		c.cacheLock.Lock()
		return c.createCache(namespace)
```

Even if T2 creates the cache and T1 enters again in the c.createCache(namespace), in that method it will detect that cache is created and it will return the correct value, but readers of this will be protected.